### PR TITLE
fixed moving index in playlist

### PR DESCRIPTION
**Pull request by sanderpol**
This pull request applies to the following story: <https://app.asana.com/0/>

## Description

**Changes proposed in this pull request (technical):**
* fixed up and down in playlist
*

## Checklist:
- [ ] :nerd_face: Documentation for this PR is covered.
- [ ] :clipboard: All tests are correct / working

### CODE REVIEW

**Review by VincentVinTin — APPROVED** [review]

File: Model/DbModels/Song.cs [hunk 1]
```diff
@@ -26,5 +27,6 @@ public class Song : DbModel
         public string PathToImage { get; set; }
 
         public SongStatus Status { get; set; }
+
```
Comment (RubenHuizenga) [inline Model/DbModels/Song.cs]: ??

File: View/DataContexts/SongListDataContext.cs [hunk 2]
```diff
@@ -32,7 +34,10 @@ public void UpdateSongInfoList()
         {
             var artistController = ArtistController.Create(DatabaseContext.Instance);
             var songController = SongController.Create(DatabaseContext.Instance);
+            if (!IsPlaylistScreen)
+                SelectedSongInfo = null;
             
+
```
Comment (RubenHuizenga) [inline View/DataContexts/SongListDataContext.cs]: ??

File: View/Screens/PlaylistScreen.xaml.cs [hunk 3]
```diff
@@ -48,25 +48,23 @@ private void RemovePlaylistButton_Click(object sender, RoutedEventArgs e)
...
-            var selectedSongInfo = (SongInfo)((ListView)mainGrid.FindName("SongList"))?.SelectedItem;
-
+            var selectedSongInfo = SongListDataContext.Instance.SelectedSongInfo;
+            
             if (selectedSongInfo == null || selectedSongInfo.Index - 1 < 0) return;
 
             playlistSongController.SwapSongs(MainWindow.CurrentPlayList, selectedSongInfo.Index, selectedSongInfo.Index - 1);
-            SongListDataContext.Instance.OnPropertyChanged("SongInfoList");
+            SongListDataContext.Instance.OnPropertyChanged("");
+
```
Comment (RubenHuizenga) [inline View/Screens/PlaylistScreen.xaml.cs:57]: ??

File: View/Styles/SongList.xaml.cs [hunk 4]
```diff
@@ -84,12 +84,19 @@ private void MenuItem_LeftClick(object sender, MouseButtonEventArgs e)
...
             SongAlterationDataContext.Instance.SetSong(songInfo.Song);
             MainWindow.InstanceMainWindow.SetScreen(ScreenNames.SongAlterationScreen);
         }
+
+        private void SelectedItem_Selected(object sender, RoutedEventArgs e )
+        {
+            var songInfo = (((ListViewItem) sender).Content as SongInfo);
+            SongListDataContext.Instance.SelectedSongInfo = songInfo;
+        }
+
```
Comment (RubenHuizenga) [inline View/Styles/SongList.xaml.cs:100]: ??

**Review by RubenHuizenga — CHANGES_REQUESTED** [review]

File: View/Styles/SongList.xaml.cs (hunk 4 above)
Comment (Rblindeboom) [reply View/Styles/SongList.xaml.cs:100]: geen enter aub.

**Review by Rblindeboom — APPROVED** [review]

**Review by RubenHuizenga — APPROVED** [review]

**Pull request merged**